Lib/test/test_unittest/testmock/testpatch.py

```diff
@@ -1979,5 +1979,86 @@ def test(): pass
             test()
 
 
+    def test_method_patched_by_subclass1(self):
+        @patch('%s.something' % __name__, 1)
+        class Foo:
+            def test(self):
+                return something
+
+        @patch('%s.something' % __name__, 2)
+        class Bar(Foo):
+            pass
+
+        self.assertEqual(Foo().test(), 1)
+        self.assertEqual(Bar().test(), 2)
+
+
+    def test_method_patched_by_subclass2(self):
+        class Foo:
+            def test(self):
+                return something
+
+        @patch('%s.something' % __name__, 2)
+        class Bar(Foo):
+            pass
+
+        self.assertEqual(Bar().test(), 2)
```

Comment:
Let's always test both classes. In this case `Foo().test()` should be unpatched.

Reply:
Right. In case of `test_method_patched_by_subclass3` though, `Foo` doesn't have a `test` method, and it doesn't make sense to instantiate the mixin class.